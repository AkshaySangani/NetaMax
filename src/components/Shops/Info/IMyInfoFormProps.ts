import { IShopUser } from "dataflows/Shops/IShopUser";

export type IMyInfoFormProps = IShopUser;
