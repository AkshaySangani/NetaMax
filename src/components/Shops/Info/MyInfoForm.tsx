import React, { ReactElement } from "react";

import { splitHost } from "utils/stringUtils";

import { FormControl, FormLabel } from "@chakra-ui/form-control";
import { Input, InputGroup } from "@chakra-ui/input";
import { Grid, GridItem } from "@chakra-ui/layout";

import { IMyInfoFormProps } from "./IMyInfoFormProps";

/**
 * The MyInfoForm component.
 * @returns {ReactElement} the MyInfoForm component.
 * @param {IMyInfoFormProps} props ProfitBanner props
 */
export const MyInfoForm = (props: IMyInfoFormProps): ReactElement => {
  const { Name, CompanyAddress, CompanyPhoneNumber, Hosts, CompanyName } = props;

  return (
    <Grid templateColumns="repeat(5, 1fr)" gap={4} ml={[0, 14]} mr={[0, 14]} mt={[4, 4]}>
      <GridItem colSpan={[12, 12, 2]} h={[16, 20]}>
        <FormControl>
          <FormLabel htmlFor="username">Nombre</FormLabel>
          <InputGroup size="md">
            <Input id="name" placeholder={CompanyName} isDisabled />
          </InputGroup>
        </FormControl>
      </GridItem>
      <GridItem colSpan={[12, 12, 2]} h={[16, 20]} colStart={[0, 0, 4]} mt={[2, 0]}>
        <FormControl>
          <FormLabel htmlFor="address">Dirección</FormLabel>
          <InputGroup size="md">
            <Input id="address" placeholder={CompanyAddress} isDisabled />
          </InputGroup>
        </FormControl>
      </GridItem>
      <GridItem colSpan={[12, 12, 2]} h={[16, 20]} mt={[2, 0]}>
        <FormControl>
          <FormLabel htmlFor="phoneNumber">Teléfono</FormLabel>
          <Input id="phoneNumber" placeholder={CompanyPhoneNumber} isDisabled />
        </FormControl>
      </GridItem>
      <GridItem colSpan={[12, 12, 2]} colStart={[0, 0, 4]} h={[16, 20]} mt={[2, 0]}>
        <FormControl>
          <FormLabel htmlFor="storeName">Nombre de la tienda</FormLabel>
          <Input id="storeName" placeholder={Name} isDisabled />
        </FormControl>
      </GridItem>
      <GridItem colSpan={[12, 12, 2]} colStart={[0, 0]} h={[16, 20]} mt={[2, 0]}>
        <FormControl>
          <FormLabel htmlFor="url">Url</FormLabel>
          <Input id="url" placeholder={`https://neta.mx?${splitHost(Hosts)}`} isDisabled />
        </FormControl>
      </GridItem>
    </Grid>
  );
};
