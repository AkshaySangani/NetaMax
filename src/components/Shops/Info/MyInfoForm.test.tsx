import { render, cleanup } from "@testing-library/react";
import { MyInfoForm } from "./MyInfoForm";
import { IMyInfoFormProps } from "./IMyInfoFormProps";

describe("MyInfoForm Component", () => {
  let mock: IMyInfoFormProps;
  beforeEach(() => {
    mock = {
      Hosts: "Host.ABC",
      CompanyPhoneNumber: "xxx xxx xxxx",
      CompanyAddress: "NetaMX Head Office",
      Name: "NetaMX",
      CompanyName: "NetaMX",
      Id: "nmx123",
    };
  });

  afterEach(() => {
    cleanup();
  });

  test("MyInfoForm should be render correctly", () => {
    const { container } = render(<MyInfoForm {...mock} />);
    expect(container).toMatchSnapshot();
  });
});
