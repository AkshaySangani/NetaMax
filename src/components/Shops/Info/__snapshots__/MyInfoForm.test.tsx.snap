// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyInfoForm Component MyInfoForm should be render correctly 1`] = `
<div>
  <div
    class="css-n178hi"
  >
    <div
      class="css-1rh2iev"
    >
      <div
        class="chakra-form-control css-0"
        role="group"
      >
        <label
          class="chakra-form__label css-uvho5g"
          for="username"
          id="field-1-label"
        >
          Nombre
        </label>
        <div
          class="chakra-input__group css-4302v8"
        >
          <input
            class="chakra-input css-0"
            disabled=""
            id="name"
            placeholder="NetaMX"
          />
        </div>
      </div>
    </div>
    <div
      class="css-vawzhs"
    >
      <div
        class="chakra-form-control css-0"
        role="group"
      >
        <label
          class="chakra-form__label css-uvho5g"
          for="address"
          id="field-2-label"
        >
          Dirección
        </label>
        <div
          class="chakra-input__group css-4302v8"
        >
          <input
            class="chakra-input css-0"
            disabled=""
            id="address"
            placeholder="NetaMX Head Office"
          />
        </div>
      </div>
    </div>
    <div
      class="css-1jpvvaf"
    >
      <div
        class="chakra-form-control css-0"
        role="group"
      >
        <label
          class="chakra-form__label css-uvho5g"
          for="phoneNumber"
          id="field-3-label"
        >
          Teléfono
        </label>
        <input
          class="chakra-input css-0"
          disabled=""
          id="phoneNumber"
          placeholder="xxx xxx xxxx"
        />
      </div>
    </div>
    <div
      class="css-vawzhs"
    >
      <div
        class="chakra-form-control css-0"
        role="group"
      >
        <label
          class="chakra-form__label css-uvho5g"
          for="storeName"
          id="field-4-label"
        >
          Nombre de la tienda
        </label>
        <input
          class="chakra-input css-0"
          disabled=""
          id="storeName"
          placeholder="NetaMX"
        />
      </div>
    </div>
    <div
      class="css-1v7sldz"
    >
      <div
        class="chakra-form-control css-0"
        role="group"
      >
        <label
          class="chakra-form__label css-uvho5g"
          for="url"
          id="field-5-label"
        >
          Url
        </label>
        <input
          class="chakra-input css-0"
          disabled=""
          id="url"
          placeholder="https://neta.mx?Host"
        />
      </div>
    </div>
  </div>
</div>
`;
